doc/rst/language/spec/arrays.rst

```diff
@@ -162,23 +162,42 @@ Rectangular Array Literals
 
 Rectangular array literals are specified by enclosing a
 comma-separated list of expressions in square brackets, where each
-expression represents an array element's value.  A trailing comma is
-permitted after the final array element.  For a literal with `n`
+expression represents an array element's value. A trailing comma is
+permitted after the final array element. For a literal with `n`
 expressions, an anonymous domain literal ``{0..<n}`` is generated to
-represent the array's indices.  If the value expressions are all of
+represent the array's indices. If the value expressions are all of
 the same type, that type will be used as the array's element type.  If
 they are not, the array's element type is computed using the same type
 unification logic that is used when inferring the return type of a
 procedure with an implicit return type (see
 :ref:`Implicit_Return_Types`).
 
-
+A multi-dimensional rectangular array literal is specified using
+comma-separated values, with each dimension separated by a semicolon. Multiple
```

Comment:
* maybe "comma-separated values in the innermost dimension"?
* "and semicolons to separate outer dimensions."?
* The "Multiple semicolons…" sentence feels a bit vague/imprecise.  I wonder whether we can just skip from the previous sentence to an example, like "For example, a 3D array literal will use commas to separate the elements in a row, a semicolon to mark the end of each row, and two consecutive semicolons to indicate the end of a plane of values."  (in which case the n and n-1 sentence could be dropped as well).

Reply:
I think having the n/n-1 sentence before an example is helpful, but I agree the example helps a lot

This now is

> A multi-dimensional rectangular array literal is specified using
comma-separated values in the innermost dimension and semicolons to separate
outer dimensions. An array literal of rank ``n`` uses ``n-1`` consecutive semicolons to
separate the last dimension. For example, a 3D array literal will use commas to
separate the elements in a row, a semicolon to mark the end of each row, and
two consecutive semicolons to indicate the end of a plane of values.

Reply:
Works for me, though in the preserved sentence, I'd suggest "outermost dimension" rather than "last dimension" (particularly since I can't put a `;;` at the end of my 3D literal, which feels like the "last" of the outermost dimensions).